client/post-editor/post-editor.jsx

```diff
@@ -525,11 +537,7 @@ export class PostEditor extends React.Component {
 		}
 	}
 
-	autosave = async () => {
-		// If debounced / throttled autosave was pending, consider it flushed
-		this.throttledAutosave.cancel();
-		this.debouncedAutosave.cancel();
-
+	async doAutosave() {
```

Comment:
I know this code was already here, but does this method work correctly? Using `isState` as a guard while setting it soon afterwards is a bit of a code smell.

Reply:
`setState` is still synchronous today, but you're right that there are potential race conditions once React concurrent mode is enabled. Then we'll need to fix it, or migrate everyone away from the Classic Editor :wink:

Reply:
Sounds reasonable enough 👍